OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseUnitMobileNotSupportedFragment.java

```diff
@@ -42,13 +46,31 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
 
         if (unit.getAuthorizationDenialReason() == AuthorizationDenialReason.FEATURE_BASED_ENROLLMENTS) {
             if (environment.getRemoteFeaturePrefs().isValuePropEnabled()) {
+                boolean isSelfPaced = getBooleanArgument(Router.EXTRA_IS_SELF_PACED, false);
+                String price = getStringArgument(Router.EXTRA_PRICE);
                 binding.containerLayoutNotAvailable.setVisibility(View.GONE);
                 binding.llGradedContentLayout.setVisibility(View.VISIBLE);
-                binding.btnLearnMore.setOnClickListener(v ->
-                        CourseModalDialogFragment.newInstance(environment.getConfig().getPlatformName(),
-                                false, unit.getCourseId(), unit.getBlockId())
-                                .show(getChildFragmentManager(), CourseModalDialogFragment.TAG));
-                environment.getAnalyticsRegistry().trackLockedContentTapped(unit.getCourseId(), unit.getBlockId());
+                binding.layoutUpgradeBtn.btnUpgrade.setVisibility(environment.getConfig().isIAPEnabled() ? View.VISIBLE : View.GONE);
+
+                binding.layoutUpgradeFeature.supportNonProfit.setText(
+                        ResourceUtil.getFormattedString(getResources(), R.string.course_modal_support_non_profit, PLATFORM_NAME, environment.getConfig().getPlatformName()));
+
+                binding.toggleShow.setOnClickListener(v -> {
+                    boolean showMore = binding.toggleShow.getText().toString().equals(getString(R.string.course_modal_graded_assignment_show_more));
+                    binding.layoutUpgradeFeature.containerLayout.setVisibility(showMore ? View.VISIBLE : View.GONE);
+                    binding.toggleShow.setText(getText(showMore ? R.string.course_modal_graded_assignment_show_less : R.string.course_modal_graded_assignment_show_more));
+                    environment.getAnalyticsRegistry().trackValuePropShowMoreLessTapped(unit.getCourseId(),
+                            price, unit.getComponentSectionId(), unit.getComponentSubsectionId(),
+                            unit.getComponentUnitId(), unit.getId(), isSelfPaced, showMore);
+                });
+
+                binding.layoutUpgradeBtn.btnUpgrade.setOnClickListener(v -> {
+                    environment.getAnalyticsRegistry().trackUpgradeNowTapped(unit.getCourseId(),
+                            price, unit.getComponentSectionId(), unit.getComponentSubsectionId(),
+                            unit.getComponentUnitId(), unit.getId(), isSelfPaced);
+                    // Waiting for the decision on this analytic
+                    //environment.getAnalyticsRegistry().trackLockedContentTapped(unit.getCourseId(), unit.getBlockId());
```

Comment:
As per discussion in the team meeting, this event should be triggered when the learner click | navigation to the locked content.